core/src/main/scala/akka/kafka/internal/KafkaConsumerActor.scala

```diff
@@ -453,7 +477,7 @@ import scala.util.control.NonFatal
     commitRefreshing.updateRefreshDeadlines(assignedOffsets.keySet)
   }
 
-  private def commit(commitMap: Map[TopicPartition, OffsetAndMetadata], reply: ActorRef): Unit = {
+  private def commit(commitMap: Map[TopicPartition, OffsetAndMetadata], sendReply: AnyRef => Unit): Unit = {
     commitRefreshing.updateRefreshDeadlines(commitMap.keySet)
     commitsInProgress += 1
     val startTime = System.nanoTime()
```

Comment:
What if commitAsync throws exception?

Reply:
That exception is passed to the call-back as parameter.